examples/ingest/github/ingest.sh

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+
+# Processes the Unstructured-IO/unstructured repository
+# through Unstructured's library in 2 processes.
+
+# Structured outputs are stored in github-ingest-output/
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+cd "$SCRIPT_DIR"/../../.. || exit 1
+
+PYTHONPATH=. ./unstructured/ingest/main.py \
+    --github-url Unstructured-IO/unstructured \
+    --github-branch main \
+    --structured-output-dir github-ingest-output \
+    --num-processes 2 \
+    --verbose
+
+# Alternatively, you can call it using:
+# unstructured-ingest --github-url ...
```

Comment:
I ran into the following error on mac. (Maybe mac only not linux?)

```
<snip>
The file 'unstructured/staging/label_box.py' is discarded as it does not contain a supported filetype.
The file 'unstructured/staging/label_studio.py' is discarded as it does not contain a supported filetype.
The file 'unstructured/staging/prodigy.py' is discarded as it does not contain a supported filetype.
The file 'unstructured/utils.py' is discarded as it does not contain a supported filetype.
Traceback (most recent call last):
  File "./unstructured/ingest/main.py", line 204, in <module>
    main()
  File "/Users/cragwolfe/.pyenv/versions/unst/lib/python3.8/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/Users/cragwolfe/.pyenv/versions/unst/lib/python3.8/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/Users/cragwolfe/.pyenv/versions/unst/lib/python3.8/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/cragwolfe/.pyenv/versions/unst/lib/python3.8/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "./unstructured/ingest/main.py", line 195, in main
    MainProcess(
  File "./unstructured/ingest/main.py", line 63, in run
    results = pool.map(self.doc_processor_fn, docs)  # noqa: F841
  File "/Users/cragwolfe/.pyenv/versions/3.8.15/lib/python3.8/multiprocessing/pool.py", line 364, in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
  File "/Users/cragwolfe/.pyenv/versions/3.8.15/lib/python3.8/multiprocessing/pool.py", line 771, in get
    raise self._value
  File "/Users/cragwolfe/.pyenv/versions/3.8.15/lib/python3.8/multiprocessing/pool.py", line 537, in _handle_tasks
    put(task)
  File "/Users/cragwolfe/.pyenv/versions/3.8.15/lib/python3.8/multiprocessing/connection.py", line 206, in send
    self._send_bytes(_ForkingPickler.dumps(obj))
  File "/Users/cragwolfe/.pyenv/versions/3.8.15/lib/python3.8/multiprocessing/reduction.py", line 51, in dumps
    cls(buf, protocol).dump(obj)
  File "/Users/cragwolfe/.pyenv/versions/3.8.15/lib/python3.8/multiprocessing/synchronize.py", line 101, in __getstate__
    context.assert_spawning(self)
  File "/Users/cragwolfe/.pyenv/versions/3.8.15/lib/python3.8/multiprocessing/context.py", line 359, in assert_spawning
    raise RuntimeError(
RuntimeError: RLock objects should only be shared between processes through inheritance
```

I have a strong suspicion it is due to `<class 'github.Repository.Repository'>` which is a member of each `GitHubIngestDoc` instance not playing well with multiprocessing.

Reply:
Please retry with `pygithub==1.57.0`. 1.58.0 indeed has a problem with multiprocessing.

Reply:
That fixes it! And I see you have it pinned in setup.py.

Can you add a `NOTE` in setup.py to change to that dep to `pygithub>1.58.0` if the future version resolves the issue?

Reply:
I appreciate your time on this as well! 😊

Reply:
Done!